src/coreclr/src/vm/gchelpers.cpp

```diff
@@ -550,6 +550,15 @@ OBJECTREF AllocateArrayEx(MethodTable *pArrayMT, INT32 *pArgs, DWORD dwNumArgs,
     }
 #endif
 
+    // IBC Log MethodTable access
+    g_IBCLogger.LogMethodTableAccess(pArrayMT);
+    SetTypeHandleOnThreadForAlloc(TypeHandle(pArrayMT));
+
+    // keep original flags in case the call is recursive (jugged array case)
+    // the aditional flags that we infer here, such as GC_ALLOC_CONTAINS_REF
+    // may not be applicable to inner arrays
+    GC_ALLOC_FLAGS flags_orig = flags;
```

Comment:
>GC_ALLOC_FLAGS flags_orig = flags; [](start = 4, length = 34)

nit - why use a GC coding style here  (instead of flagsOriginal)? 

you are changing this functionally so that's a little concerning but I agree that this is the way it should work so hopefully we didn't have scenarios the original way happened to work, and if so I hope we have tests to catch it.

Reply:
Should change to `flagsOriginal`

Reply:
I think it was incorrect, but not dangerous. Setting CONTAINS_REF incorrectly for the inner array would not have much effect. I am not sure it effects anything at lower layers.
Now, we would also propagate LARGE_OBJECT_HEAP to the inner, if the outer is large. That would be more observable, so I decided to change this.

---
In reply to: [393958807](https://github.com/dotnet/runtime/pull/33526#discussion_r393958807) [](ancestors = 393958807)